src/pl/michal/interfejs/MySQLaccess.java

```diff
@@ -0,0 +1,202 @@
+            accountBalance = 0.0;
+        if (accountLimit < 0) {
+            accountLimit = 0;
+        }
+        try {
+            preparedStatement = connection.prepareStatement("INSERT INTO accountData VALUES (?,?,?,null)");
+            preparedStatement.setInt(1, accountNO);
+            preparedStatement.setDouble(2, accountBalance);
+            preparedStatement.setInt(3, accountLimit);
+            preparedStatement.executeUpdate();
+        } catch (Exception e) {
+            try {
+                if (connection != null)
+                    connection.rollback();
+            } catch (Exception ex) {
+                throw ex;
+            }
+            throw e;
+        }
+    }
+
+    public void FinishTransaction() throws Exception {
+        try {
+            if (connection != null)
+                connection.commit();
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public void CloseDB() throws Exception {
+        try {
+            if (resultSet != null)
+                resultSet.close();
+            if (preparedStatement != null)
+                preparedStatement.close();
+            if (statement != null)
+                statement.close();
+            if (connection != null)
+                connection.close();
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public double GetAccountInfo(int accountNO) throws Exception {
+        if (accountNO <= 0) {
+            throw new IllegalArgumentException("No passed Account number");
+        }
+
+        try {
+            statement = connection.createStatement();
+//            resultSet=statement.executeQuery(String.format("SELECT accountBalance FROM accountData WHERE accountNO=%d",accountNO));
+            resultSet = statement.executeQuery("SELECT accountBalance FROM accountData WHERE accountNO=" + accountNO);
+            if (resultSet.next()) {
+                return resultSet.getDouble(1);
+            }
+            return - 1;
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public void UpdateAccountData(int accountNO, double val, int transferID) throws Exception {
+        if (accountNO <= 0 || val < 0 || transferID <= 0) {
+            throw new IllegalArgumentException("Wrong argument data\n");
+        }
+        String sqlUpdate = "UPDATE accountData "
+                + "SET accountBalance = ?,"
+                + "bankTransaction_transactionID = ? "
+                + "WHERE accountNO = ?";
+        try {
+            preparedStatement = connection.prepareStatement(sqlUpdate);
+            preparedStatement.setDouble(1, val);
+            preparedStatement.setInt(2, transferID);
+            preparedStatement.setInt(3, accountNO);
+            preparedStatement.executeUpdate();
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public void InsertBankTransaction(int from, int to, double amount, int fee) throws Exception {
```

Comment:
Dobra praktyka: zrób sobie nawyk dodawania komentarzy do metod, nie musisz nawet opisywać argumentów. Ale w takim Intellij wystarczy że nad metodą zaczniesz wpisywać /**
Ja czytając teraz kod muszę przeglądać kod zastanawiając się co to jest 'from' oraz 'to' - jakieś int'y ale co dokładnie to nie wiadomo.